File: course/src/main/java/in/testpress/course/ui/fragments/content_detail_fragments/AttachmentContentFragment.java
package in.testpress.course.ui.fragments.content_detail_fragments;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.TextView;

import in.testpress.core.TestpressSDKDatabase;
import in.testpress.core.TestpressSdk;
import in.testpress.course.R;
import in.testpress.models.greendao.Attachment;
import in.testpress.models.greendao.AttachmentDao;
import in.testpress.models.greendao.Content;
import in.testpress.models.greendao.CourseAttempt;
import in.testpress.util.ViewUtils;

public class AttachmentContentFragment extends BaseContentDetailFragment {

    private LinearLayout attachmentContentLayout;
    private AttachmentDao attachmentDao;
    private TextView titleView;
    private LinearLayout titleLayout;
    
    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        attachmentDao = TestpressSDKDatabase.getAttachmentDao(getActivity());
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.attachment_content_detail, container, false);
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        attachmentContentLayout = (LinearLayout) view.findViewById(R.id.attachment_content_layout);
        titleView = (TextView) view.findViewById(R.id.title);
        titleLayout = (LinearLayout) view.findViewById(R.id.title_layout);


        if (content != null) {
            loadContent();
        }
    }

    @Override
    void loadContent() {
        titleView.setText(content.getTitle());
        titleLayout.setVisibility(View.VISIBLE);

        if (content.getRawAttachment() == null) {
            updateContent();
            return;
        }
        TextView description = (TextView) getView().findViewById(R.id.attachment_description);
        final Attachment attachment = content.getRawAttachment();
        if (attachment.getDescription() != null && !attachment.getDescription().isEmpty()) {
            description.setText(attachment.getDescription());
            description.setTypeface(TestpressSdk.getRubikRegularFont(getActivity()));
            description.setVisibility(View.VISIBLE);
        } else {
            description.setVisibility(View.GONE);
        }
        Button downloadButton = (Button) getView().findViewById(R.id.download_attachment);
        ViewUtils.setLeftDrawable(getActivity(), downloadButton, R.drawable.ic_file_download_18dp);
        downloadButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(Intent.ACTION_VIEW,
                        Uri.parse(attachment.getAttachmentUrl())));
            }
        });
        attachmentContentLayout.setVisibility(View.VISIBLE);
        swipeRefresh.setRefreshing(false);
        createContentAttempt();
    }

    @Override
    void onUpdateContent(Content fetchedContent) {
        Attachment attachment = fetchedContent.getRawAttachment();
        if (attachment != null) {
            attachmentDao.insertOrReplace(attachment);
            fetchedContent.setAttachmentId(attachment.getId());
        }
        contentDao.insertOrReplace(fetchedContent);
        content = fetchedContent;
    }

    @Override
    void onCreateContentAttempt(CourseAttempt courseAttempt) {}

    @Override
    void hideContents() {
        attachmentContentLayout.setVisibility(View.GONE);
    }
}

File: course/src/main/java/in/testpress/course/ui/fragments/content_fragments/AttachmentContentFragment.kt
package `in`.testpress.course.ui.fragments.content_fragments

import `in`.testpress.core.TestpressSDKDatabase
import `in`.testpress.core.TestpressSdk
import `in`.testpress.course.R
import `in`.testpress.models.greendao.AttachmentDao
import `in`.testpress.models.greendao.Content
import `in`.testpress.util.ViewUtils
import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.LinearLayout
import android.widget.TextView


class AttachmentContentFragment : BaseContentDetailFragment() {

    private lateinit var attachmentContentLayout: LinearLayout
    private lateinit var titleView: TextView
    private lateinit var description: TextView
    private lateinit var titleLayout: LinearLayout
    private lateinit var downloadButton: Button

    private lateinit var attachmentDao: AttachmentDao

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        attachmentDao = TestpressSDKDatabase.getAttachmentDao(activity)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.attachment_content_detail, container, false);
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        attachmentContentLayout = view.findViewById(R.id.attachment_content_layout)
        titleView = view.findViewById(R.id.title)
        titleLayout = view.findViewById(R.id.title_layout)
        description = view.findViewById(R.id.description)
        downloadButton = view.findViewById(R.id.download_attachment)
        ViewUtils.setTypeface(arrayOf(titleView), TestpressSdk.getRubikMediumFont(activity!!))
        ViewUtils.setLeftDrawable(context, downloadButton, R.drawable.ic_file_download_18dp)
    }

    override fun loadContent() {
        titleView.text = content.title
        titleLayout.visibility = View.VISIBLE

        if (content.rawAttachment == null) {
            updateContent()
            return
        }

        val attachment = content.rawAttachment
        if (!attachment.description.isNullOrEmpty()) {
            description.text = attachment.description
            description.typeface = TestpressSdk.getRubikRegularFont(context!!)
        }

        downloadButton.setOnClickListener() {
            context!!.startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(attachment.attachmentUrl)))
        }
        attachmentContentLayout.visibility = View.VISIBLE
        viewModel.createContentAttempt()
    }

    override fun onUpdateContent(content: Content) {
        val attachment = content.rawAttachment
        contentDao.deleteAll()
        attachment?.let {
            attachmentDao.insertOrReplaceInTx(it)
            content.attachmentId = it.id
            contentDao.delete(content)
            contentDao.insertOrReplaceInTx(content)
        }
    }
}
